File: expresslk-webportal/module/Api/src/Api/Client/Soap/Core/SupplierAccount.php
<?php

namespace Api\Client\Soap\Core;

include_once('Entity.php');

class SupplierAccount extends Entity
{

    /**
     * @var int $allowedDivisions
     * @access public
     */
    public $allowedDivisions = null;

    /**
     * @var string $bank
     * @access public
     */
    public $bank = null;

    /**
     * @var string $branch
     * @access public
     */
    public $branch = null;

    /**
     * @var int $id
     * @access public
     */
    public $id = null;

    /**
     * @var boolean $isPrimary
     * @access public
     */
    public $isPrimary = null;

    /**
     * @var string $name
     * @access public
     */
    public $name = null;

    /**
     * @var string $number
     * @access public
     */
    public $number = null;

    /**
     * @var int $supplierId
     * @access public
     */
    public $supplierId = null;

    /**
     * @var string $swift
     * @access public
     */
    public $swift = null;

    /**
     * @var AccountType $type
     * @access public
     */
    public $type = null;

    /**
     * @param int $allowedDivisions
     * @param string $bank
     * @param string $branch
     * @param int $id
     * @param boolean $isPrimary
     * @param string $name
     * @param string $number
     * @param int $supplierId
     * @param string $swift
     * @param AccountType $type
     * @access public
     */
    public function __construct($allowedDivisions, $bank, $branch, $id, $isPrimary, $name, $number, $supplierId, $swift, $type)
    {
      $this->allowedDivisions = $allowedDivisions;
      $this->bank = $bank;
      $this->branch = $branch;
      $this->id = $id;
      $this->isPrimary = $isPrimary;
      $this->name = $name;
      $this->number = $number;
      $this->supplierId = $supplierId;
      $this->swift = $swift;
      $this->type = $type;
    }

}

File: expresslk-services/ExpressLK-core/src/main/java/lk/express/supplier/SupplierAccount.java
package lk.express.supplier;

import java.math.BigInteger;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.xml.bind.annotation.XmlRootElement;
import javax.xml.bind.annotation.XmlType;

import lk.express.bean.HasAllowedDivisions;

import org.hibernate.annotations.Type;

@Entity
@Table(name = "supplier_account")
@XmlType(name = "SupplierAccount", namespace = "http://supplier.express.lk")
@XmlRootElement
public class SupplierAccount extends lk.express.bean.Entity implements HasAllowedDivisions {

	private static final long serialVersionUID = 1L;

	@XmlType(name = "AccountType", namespace = "http://supplier.express.lk")
	public enum AccountType {
		Savings, Current;
	}

	@Id
	@GeneratedValue
	private Integer id;
	@Column(name = "supplier_id")
	private Integer supplierId;
	@Column(name = "number")
	private String number;
	@Column(name = "name")
	private String name;
	@Column(name = "bank")
	private String bank;
	@Column(name = "branch")
	private String branch;
	@Column(name = "swift")
	private String swift;
	@Column(name = "type")
	@Enumerated(EnumType.STRING)
	private AccountType type;
	@Column(name = "is_primary", columnDefinition = "bit")
	@Type(type = "org.hibernate.type.NumericBooleanType")
	private Boolean isPrimary;

	@Column(name = "allowed_divisions")
	private BigInteger allowedDivisions;

	@Override
	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public Integer getSupplierId() {
		return supplierId;
	}

	public void setSupplierId(Integer supplierId) {
		this.supplierId = supplierId;
	}

	public String getNumber() {
		return number;
	}

	public void setNumber(String number) {
		this.number = number;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getBank() {
		return bank;
	}

	public void setBank(String bank) {
		this.bank = bank;
	}

	public String getBranch() {
		return branch;
	}

	public void setBranch(String branch) {
		this.branch = branch;
	}

	public String getSwift() {
		return swift;
	}

	public void setSwift(String swift) {
		this.swift = swift;
	}

	public AccountType getType() {
		return type;
	}

	public void setType(AccountType type) {
		this.type = type;
	}

	public Boolean getIsPrimary() {
		return isPrimary;
	}

	public void setIsPrimary(Boolean isPrimary) {
		this.isPrimary = isPrimary;
	}

	@Override
	public BigInteger getAllowedDivisions() {
		return allowedDivisions;
	}

	@Override
	public void setAllowedDivisions(BigInteger allowedDivisions) {
		this.allowedDivisions = allowedDivisions;
	}
}
